docs/explanation/dss-arch.rst

```diff
@@ -0,0 +1,153 @@
+DSS relies on MicroK8s, a lightweight Kubernetes distribution.
+
+Therefore, MicroK8s needs to be deployed before installing DSS on the host machine. 
+It must be configured with the storage add-on. 
+This is required to use Hostpath storage in the cluster. 
+
+DSS can run with or without the use of GPUs.
+If :ref:`gpu_support` is needed, MicroK8s can be configured with the desired `GPU add-on <https://microk8s.io/docs/addon-gpu>`_. 
+
+.. _gpu_support:
+
+GPU support
+^^^^^^^^^^^
+
+DSS does not automatically install the tools and libraries required for running GPU workloads.
+To do so, it relies on MicroK8s, or any other Kubernetes deployment used as the orchestration solution.
+
+DSS is designed to support the deployment of containerised GPU workloads on NVIDIA GPUs. 
+MicroK8s simplifies the GPU access and usage through the `NVIDIA GPU Operator <NVIDIA Operator_>`_. 
+
+.. caution::
+   GPUs from other silicon vendors rather than NVIDIA can be configured. However, its functionality is not guaranteed.
+ 
+Storage
+^^^^^^^
+
+DSS expects a default `storage class <https://kubernetes.io/docs/concepts/storage/storage-classes/>`_ in the Kubernetes deployment, which is used to persist Jupyter Notebooks and MLflow artefacts.   
+In MicroK8s, the Hostpath storage add-on is chosen, used to provision Kubernetes' *PersistentVolumeClaims* (`PVCs <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_). 
+
+A shared PVC is used across all Jupyter Notebooks to share and persist data. 
+MLflow also uses its dedicated PVC to store the logged artefacts.
+This is the DSS default storage configuration and cannot be altered.
+
+This choice ensures that all storage is backed up on the host machine in the event of MicroK8s restarts.
+
+.. note::
+   By default, users can access this storage anytime under `/var/snap/microk8s/common/default-storage`. 
+
+The following diagram summarises the DSS storage:
+
+.. figure:: https://assets.ubuntu.com/v1/2130fd66-dss_storage.png
+   :width: 800px
+   :alt: text
+
+   Storage overview
+
+.. _os_layer:
+
+Operating system
+~~~~~~~~~~~~~~~~
+
+DSS is native on Ubuntu, being developed, tested and validated on it. 
+
+Moreover, the solution can be used on any Linux distribution, Windows through Windows Subsystem Linux (WSL) or MacOS through Multipass, 
+but the GPU support is at risk, depending on the GPU operator. 
+
+Namespace configuration
+-----------------------
+
+DSS has a dedicated namespace running on MicroK8s. 
+By default, it has two Kubernetes pods, but users can add new pods or attach new ML frameworks to the existing ones. 
+
+The NVIDIA GPU support runs on a dedicated namespace. 
+This includes the GPU Operator for managing access and usage, together with the `Network Operator <https://docs.nvidia.com/networking/display/cokan10/network+operator>`_, for managing networking components and enabling fast networking. 
+
+.. figure:: https://assets.ubuntu.com/v1/d28ea080-dss_namespace_config.png
```

Comment:
This diagram is not very clear IMO. I do not understand what we want to say with the right part (rhomboid)? Also, we have twice `snap` - what we want to say there?

Reply:
Also, this diagram has different color palette than the others. I have been tweaking them here: https://docs.google.com/presentation/d/1il8g3dsmX5Pso8iq4ggkNyqPVz-MJ2INKLzpXg8NLgs